### PR TITLE
feat(AccessCode): add /access_codes/delete

### PR DESCRIPTION
**Pull request by mikewuu**
Required for https://github.com/seamapi/react/issues/327.

### CODE REVIEW

File: test/api/access_codes/delete.test.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import test, { type ExecutionContext } from "ava"
+
+import { getTestServer } from "fixtures/get-test-server.ts"
+
+test("POST /access_codes/update", async (t: ExecutionContext) => {
```
Comment (razor-x) [inline test/api/access_codes/delete.test.ts:5]:
```suggestion
test("DELETE /access_codes/delete", async (t: ExecutionContext) => {
```

File: test/api/access_codes/delete.test.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    {
+      headers: {
+        Authorization: `Bearer ${seed.ws2.cst}`,
+      },
+    }
+  )
+
+  t.is(access_code.code, "1234")
+
+  const res = await axios.post(
```
Comment (razor-x) [inline test/api/access_codes/delete.test.ts:27]:
```suggestion
  const res = await axios.delete(
```

**Review by razor-x — CHANGES_REQUESTED** [review]

**Comment by razor-x** [comment]
@mikewuu Can you update from main?

File: test/api/access_codes/delete.test.ts (hunk 2 above)
Comment (mikewuu) [reply test/api/access_codes/delete.test.ts:27]: Aren't we doing `post` for everything?

**Comment by mikewuu** [comment]
@razor-x done! 👍

File: test/api/access_codes/delete.test.ts (hunk 2 above)
Comment (razor-x) [reply test/api/access_codes/delete.test.ts]: No, but I realize this is confusing. POST is supported for all endpoints, but this is just a convenience. The SDK will always use the more semantic method (and we will use that in the docs as well).

Comment (mikewuu) [reply test/api/access_codes/delete.test.ts]: gotcha thanks for clarifying

File: src/pages/api/access_codes/delete.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+const json_body = z.object({
+  access_code_id: z.string(),
+  device_id: z.string().optional(),
+  sync: z.boolean().default(false),
+})
+
+export default withRouteSpec({
+  auth: "cst_ak_pk",
+  methods: ["POST"],
```
Comment (razor-x) [inline src/pages/api/access_codes/delete.ts:15]:
```suggestion
  methods: ["POST", "DELETE"],
```

**Review by razor-x — APPROVED** [review]
Just the last suggestion with hopefully fixes the tests.

File: test/api/access_codes/delete.test.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  t.is(access_code.code, "1234")
+
+  const res = await axios.delete(
+    "/access_codes/delete",
+    {
+      access_code_id: access_code.access_code_id,
+    },
+    {
+      headers: {
+        Authorization: `Bearer ${seed.ws2.cst}`,
```
Comment (razor-x) [inline test/api/access_codes/delete.test.ts:34]:
```suggestion
      data: {
        access_code_id: access_code.access_code_id,
      },
      headers: {
        Authorization: `Bearer ${seed.ws2.cst}`,
      }
```

**Pull request merged**